libdnf/rpm/solv/goal_private.cpp

```diff
@@ -294,9 +293,8 @@ libdnf::GoalProblem GoalPrivate::resolve() {
 
     // Removal of SOLVER_WEAK to allow report errors
     if (remove_solver_weak) {
-        auto elements = job.data();
-        for (int i = 0; i < job.size(); i += 2) {
-            elements[i] &= ~SOLVER_WEAK;
+        for (auto & item : job) {
```

Comment:
The original code modified only the second (i += 2) elements. Is new code equivalent?